### PR TITLE
Convert double array to int array

### PR DESCRIPTION
**Pull request by TomFischer**
Needed sometimes for model preparation.

### CODE REVIEW

File: Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    TCLAP::CmdLine cmd(
+        "Converts a double or floating point data array of a vtk unstructured "
+        "grid into a int or double data array.",
+        ' ',
+        "0.1");
```
Comment (endJunction) [inline Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp]: See other tools for version message with copyright and build info.

File: Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/writeMeshToFile.h"
+
+#include "MeshLib/Mesh.h"
+
+template <typename ORIG, typename NEW_T>
```
Comment (endJunction) [inline Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp]: The templates' names are not so explanatory. Maybe even T1, T2 would be better.

File: Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+void castPropertyVectorToPropertyVector(
+    MeshLib::Properties& properties,
+    std::string const& property_vector_name_in,
+    std::string const& property_vector_name_out)
+{
+    auto const* const orig_property = properties.getPropertyVector<ORIG>(
+        property_vector_name_in, MeshLib::MeshItemType::Cell, 1);
+    auto* new_pv = properties.createNewPropertyVector<NEW_T>(
```
Comment (endJunction) [inline Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp]: Please check for nullptr before access. Also for orig_property.

File: Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        if (mesh->getProperties().existsPropertyVector<float>(
+                property_arg.getValue(), MeshLib::MeshItemType::Cell, 1))
+            castPropertyVectorToPropertyVector<float, double>(
+                mesh->getProperties(),
+                property_arg.getValue(),
+                new_property_arg.getValue());
+    }
+
+        MeshLib::IO::writeMeshToFile(*mesh, out_mesh_arg.getValue());
```
Comment (endJunction) [inline Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp]: indentation.

File: Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp (hunk 3 above)
Comment (TomFischer) [reply Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp]: 'getOrCreateMeshProperty' needs a mesh, but I unpacked the 'Properties' object from the mesh outside of this function and as a consequence I haven't the mesh here available.

File: web/content/docs/tools/model-preparation/convert-data-array-to-data-array/index.pandoc [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+the geometrical and the process data information is stored in one file - in so
+called data arrays.
+
+Some tools, for instance paraview, export data arrays always using a floating
+point data type. OpenGeoSys expects the 'MaterialIDs' data array to have int
+data-type.
+
+Other tools, for instance Gocad, export data associated with cells or nodes
+sometimes as float. The tool can convert the data arrays to double data-type.
+
```
[the remark this answers is not in the document]
Comment (TomFischer) [reply web/content/docs/tools/model-preparation/convert-data-array-to-data-array/index.pandoc]: Updated the docu.

File: Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    TCLAP::CmdLine cmd(
+        "Converts a double or floating point data array of a vtk unstructured "
+        "grid into a int or double data array.",
+        ' ',
+        "0.1");
+
+    TCLAP::ValueArg<std::string> new_property_data_type_arg(
+        "t",
+        "new-property-data-type",
+        "the name of the data type as string (for instance int or double)",
```
[the remark this answers is not in the document]
Comment (TomFischer) [reply Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp]: Dropped.

**Pull request merged**